pantheon-bundle/src/main/java/com/redhat/pantheon/servlet/util/VersionUploadOperation.java

```diff
@@ -0,0 +1,161 @@
+package com.redhat.pantheon.servlet.util;
+
+import com.google.common.hash.HashCode;
+import com.redhat.pantheon.asciidoctor.AsciidoctorService;
+import com.redhat.pantheon.conf.GlobalConfig;
+import com.redhat.pantheon.jcr.JcrResources;
+import com.redhat.pantheon.model.HashableFileResource;
+import com.redhat.pantheon.model.api.Child;
+import com.redhat.pantheon.model.api.SlingModels;
+import com.redhat.pantheon.model.document.Document;
+import com.redhat.pantheon.model.document.DocumentLocale;
+import com.redhat.pantheon.model.document.DocumentMetadata;
+import com.redhat.pantheon.servlet.ServletUtils;
+import org.apache.commons.lang3.LocaleUtils;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.resource.ModifiableValueMap;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.servlets.post.AbstractPostOperation;
+import org.apache.sling.servlets.post.PostResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.jcr.Session;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.Calendar;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+public abstract class VersionUploadOperation extends AbstractPostOperation {
+
+    private static final Logger log = LoggerFactory.getLogger(VersionUploadOperation.class);
+
+    private static final Set<String> METADATA_COPY_EXCLUDES = Collections.unmodifiableSet(
+            new HashSet<>(
+                    Arrays.asList(
+                            "jcr:description",
+                            "jcr:lastModified",
+                            "jcr:primaryType",
+                            "jcr:title",
+                            "pant:dateUploaded",
+                            "pant:datePublished"
```

Comment:
should we consider the following two to be added to this list?
"pant:headline"
"pant:abstract"

Reply:
Probably! If those are both populated automatically from the doc, then it doesn't make sense to carry them forward.